stream-chat-android-client/src/main/java/io/getstream/chat/android/client/extensions/StringExtensions.kt

```diff
@@ -57,3 +65,150 @@ public fun String.cidToTypeAndId(): Pair<String, String> {
     check(':' in this) { "cid needs to be in the format channelType:channelId. For example, messaging:123" }
     return checkNotNull(split(":").takeIf { it.size >= 2 }?.let { it.first() to it.last() })
 }
+
+/**
+ * Returns [StreamCdnOriginalImageDimensions] if the image is hosted by Stream's CDN and is resizable,
+ * otherwise returns null.
+ *
+ * @return Class containing the original width and height dimensions of the image or null.
+ */
+public fun String.getStreamCdnHostedImageDimensions(): StreamCdnOriginalImageDimensions? {
```

Comment:
Should it only works with Stream's CDN?
Are `ow` and `oh` "Standard" param names?
Should we use them if they come from a different host? (eg: `https://www.jc-cdn.com/picture.jpg?ow=15&oh=30`)

Reply:
This was discussed internally, but to keep everyone on the same page I'll write points here summarizing why it's believed this logic should be domain agnostic:

- Possibility of introducing custom domains in the future
- If a 3rd party CDN does serve `ow` and `oh` parameters in their links and we append resize parameters, these should most likely simply be ignored by the 3rd party CDN